client/js/components/statement/listStatements/ListStatements.vue

```diff
@@ -451,22 +426,34 @@ export default {
     },
 
     exportRoute: function () {
-      return (exportRoute) => Routing.generate(exportRoute, {
-        filter: {
-          procedureId: {
-            condition: {
-              path: 'procedure.id',
-              value: this.procedureId
+      return (exportRoute, docxHeaders) => {
+        const parameters = {
+          filter: {
+            procedureId: {
+              condition: {
+                path: 'procedure.id',
+                value: this.procedureId
+              }
             }
+          },
+          procedureId: this.procedureId,
+          search: {
+            value: this.searchValue,
+            ...this.searchFieldsSelected !== null ? { fieldsToSearch: this.searchFieldsSelected } : {}
+          },
+          sort: this.selectedSort
+        }
+
+        if (docxHeaders) {
+          parameters.tableHeaders = {
+            c_left: docxHeaders.left,
+            c_middle: docxHeaders.middle,
+            c_right: docxHeaders.right
```

Comment:
if its not too complicated to refactor, can we just use numbers for the cols. Than we can simpler extend if there will come more or we want to use the logic in other situations.
```suggestion
            col1: docxHeaders[0],
            col2: docxHeaders[1],
            col3: docxHeaders[2]
```

Reply:
We renamed it to `col1`, `col2`, and `col3`. it could be easier to handle in the future, you are right. https://github.com/demos-europe/demosplan-core/pull/3315/commits/540aa687fc1655e7584f92e760a451857def3161

However, why do you prefer an array in this case and not an object? I think it is clearer to use 'docxHeaders' as a  `key: value`